File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    
    // 1. 옷을 종류별로 구분하기
    unordered_map<string, int> map;
    for (vector<string> cloth : clothes)
        map[cloth[1]]++;
    
    // 2. 옷을 입지 않는 경우도 포함한 모든 경우의 수 계산
    int answer = 1;
    for (const auto& it : map)
        answer *= it.second +1;
    
    // 3. 전체 집합 - (한 번도 옷을 입지 않는 경우)
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    
    # 1. 옷을 종류별로 구분하기
    hash_map = {}
    for cloth, type in clothes:
        hash_map[type] = hash_map.get(type, 0) + 1
        
    # 2. 입지 않는 경우를 추가하여 모든 조합 계산
    answer = 1
    for type in hash_map:
        answer *= (hash_map[type] + 1)
    
    # 3. 아무 종류의 옷도 입지 않은 경우 제외
    return answer - 1
